tests/aws/aio/aiomoto_services.py

```diff
@@ -0,0 +1,145 @@
+import asyncio
+import functools
+import logging
+import socket
+import threading
+import time
+import os
+
+# Third Party
+import aiohttp
+import moto.backends
+import moto.server
+import werkzeug.serving
+
+
+HOST = "127.0.0.1"
+
+_PYCHARM_HOSTED = os.environ.get("PYCHARM_HOSTED") == "1"
+CONNECT_TIMEOUT = 90 if _PYCHARM_HOSTED else 10
+
+
+def get_free_tcp_port(release_socket: bool = False):
+    sckt = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sckt.bind(("", 0))
+    addr, port = sckt.getsockname()
+    if release_socket:
+        sckt.close()
+        return port
+
+    return sckt, port
+
+
+class MotoService:
+    """ Will Create MotoService.
+    Service is ref-counted so there will only be one per process. Real Service will
+    be returned by `__aenter__`."""
+
+    _services = dict()  # {name: instance}
+
+    def __init__(self, service_name: str, port: int = None):
+        self._service_name = service_name
+
+        if port:
+            self._socket = None
+            self._port = port
+        else:
+            self._socket, self._port = get_free_tcp_port()
+
+        self._thread = None
+        self._logger = logging.getLogger("MotoService")
+        self._refcount = None
+        self._ip_address = HOST
+        self._server = None
+
+    @property
+    def endpoint_url(self):
+        return "http://{}:{}".format(self._ip_address, self._port)
+
+    def reset(self):
```

Comment:
each instance should not affect other instances, if they are affecting other instances then we should run each instance in a separate process...however, the problem with that is that some services interact with one another like SNS + lambdas.  I think each test needs a process, where all moto services for that test are run in that process.  So I think you would want the process to be in pytest nomenclature scoped to the function.

Reply:
Instead this probably should just reset the service named after itself automatically during shutdown

Reply:
Without understanding all the implications, this just makes it an option without imposing it until a fixture instantiates a service.  There are ways to skip the dispatch app-generator, but AFAICT all those options also use the global BACKENDS.  There might be ways to specify both the service-name and a region to instantiate a new model for that service without adding it to BACKENDS, but didn't get that far.

Reply:
See `test_aio_aws_client` and the fixture it uses, which returns a client generator.  The `test_aio_aws_client` could be easily parameterized to run all available clients if it has some generic assertions.